app/backend/src/couchers/servicers/events.py

```diff
@@ -1043,13 +1048,19 @@ def ListMyEvents(self, request, context, session):
             occurrences = occurrences.where(EventOccurrence.end_time < page_token + timedelta(seconds=1)).order_by(
                 EventOccurrence.start_time.desc()
             )
-
-        occurrences = occurrences.limit(page_size + 1)
+        # Count the total number of items for pagination
+        total_items = session.execute(select(func.count()).select_from(occurrences.subquery())).scalar()
+        # Apply pagination by page number
+        occurrences = (
+            occurrences.offset(offset).limit(page_size) if request.page_number else occurrences.limit(page_size + 1)
+        )
         occurrences = session.execute(occurrences).scalars().all()
 
         return events_pb2.ListMyEventsRes(
             events=[event_to_pb(session, occurrence, context) for occurrence in occurrences[:page_size]],
             next_page_token=str(millis_from_dt(occurrences[-1].end_time)) if len(occurrences) > page_size else None,
+            page_number=page_number if request.page_number else None,
+            total_items=total_items if request.page_number else None,
```

Comment:
Please return this regardless of if `page_number` is set, given that it's computed anyway.